test/e2e/framework.go

```diff
@@ -57,12 +59,23 @@ type Framework struct {
 func NewFramework(baseName string) *Framework {
 	f := &Framework{
 		BaseName:                 baseName,
-		addonResourceConstraints: make(map[string]resourceConstraint),
+		AddonResourceConstraints: make(map[string]resourceConstraint),
+		NsCreateFunc:             createTestingNS,
 	}
+	return f.InstallGinkgoHooks()
+}
 
+// InstallGinkgoHooks installs the BeforeEach and AfterEach Ginkgo blocks
+func (f *Framework) InstallGinkgoHooks() *Framework {
 	BeforeEach(f.beforeEach)
 	AfterEach(f.afterEach)
+	return f
+}
 
+// SetNsCreateFunc allows to override the default namespace creation function
+// This function must be called before InstallGinkgoHooks() to take effect.
```

Comment:
actually, I don't think it's necessary that you call this before `InstallGinkgoHooks` is called. `f.beforeEach` and `f.afterEach` aren't called until Ginkgo starts running test cases, which will occur after all the initial framework initialization is done. (Calling Ginkgo's `BeforeEach` and `AfterEach` just registers those handlers.)

so doing
```go
f := NewFramework("foo")
f.NsCreateFunc = myfunc
```

should work as intended.